### PR TITLE
Google adwords conversion

### PR DESCRIPTION
**Pull request by salimhb**
rough implementation based on code generated by google adwords and here: https://support.google.com/adwords/answer/1722054

P.S.: this is **not** an implementation of the google adwords api.

### CODE REVIEW

File: lib/rack/tracker/google_adwords_conversion/google_adwords_conversion.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  end
+
+  self.position = :body
+
+  def render
+    Tilt.new( File.join( File.dirname(__FILE__), 'template', 'google_adwords_conversion.erb') ).render(self)
+  end
+
+  def self.track(name, *event)
+    { name.to_s => [event.last.deep_stringify_keys.merge('class_name' => event.first.to_s.capitalize)] }
```
Comment (jhilden) [inline lib/rack/tracker/google_adwords_conversion/google_adwords_conversion.rb]: the `deep_stringify_keys` is no longer necessary with the new centralized way of doing this

File: spec/handler/google_adwords_conversion_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    describe "default" do
+      def env
+        {'tracker' => {
+          'google_adwords_conversion' => [
+            { 'class_name' => 'Conversion', 'id' => 123456, 'language' => 'en', 'format' => '3', 'color' => 'ffffff', 'label' => 'Conversion Label' }
+          ]
+        }}
+      end
+
+      subject { described_class.new(env, tracker: 'somebody').render }
```
Comment (jhilden) [inline spec/handler/google_adwords_conversion_spec.rb]: `tracker: 'somebody'` is unnecessary, or?  could also be just `{}`

File: spec/integration/google_adwords_conversion_integration_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    setup_app(action: :google_adwords_conversion) do |tracker|
+      tracker.handler :google_adwords_conversion
+    end
+    visit '/'
+  end
+
+  subject { page }
+
+  it "embeds the script tag with tracking event from the controller action" do
+    expect(page.find("body")).to have_content("var google_conversion_id = 123456;\nvar google_conversion_language = 'en';\nvar google_conversion_format = '3';\nvar google_conversion_color = 'ffffff';\nvar google_conversion_label = 'Conversion Label';")
```
Comment (jhilden) [inline spec/integration/google_adwords_conversion_integration_spec.rb]: i would make individual assertions out of that for readability.

should there maybe a testcase with an escaped special character label?

**Comment by salimhb** [comment]
merging the integration branch here was not a good idea :(
New PR: https://github.com/railslove/rack-tracker/pull/24

**Pull request closed without merging**